efel/cppcore/Utils.cpp

```diff
@@ -200,6 +200,31 @@ std::pair<size_t, size_t> get_time_index(const std::vector<double>& t,
   return std::pair<size_t, size_t>(startIndex, endIndex);
 }
 
+// expects an empty newpeakindices
+int peaks_after_stim_start(const double stimstart,
+                           const vector<int>& peakindices,
+                           const vector<double>& t,
+                           vector<int>& newpeakindices){
+  for (size_t i=0; i < peakindices.size(); i++) {
+    if (t[peakindices[i]] > stimstart) {
+      newpeakindices.push_back(peakindices[i]);
+    }
+  }
+  return 1;
+}
```

Comment:
Thanks, since C++11 this can be shortened as follows. The compiler applies [return value optimisation](https://en.cppreference.com/w/cpp/language/copy_elision) to avoid copy/move operations.


```C++
vector<int> peaks_after_stim_start(const double stimstart,
                                   const vector<int>& peakindices,
                                   const vector<double>& t) {
    vector<int> newpeakindices;
    for (size_t i = 0; i < peakindices.size(); i++) {
        if (t[peakindices[i]] > stimstart) {
            newpeakindices.push_back(peakindices[i]);
        }
    }
    return newpeakindices;
}
```

and the usage can be simplified like this
```C++
vector<int> newpeakindices = peaks_after_stim_start(stimstart, peakindices, t);
```

Reply:
done in latest commit